TISServices/TISUnitTests/AssessmentControllerIntegrationTest.cs

```diff
@@ -0,0 +1,177 @@
+                {
+                    throw new FileNotFoundException(String.Format("Could not find scoring package XML at {0}\\{1}", Directory.GetCurrentDirectory(), testPackageFiles[i]));
+                }
+            }
+
+            // Load the XML for each file into the database.  If any of the files fail to load, the entire transaction will be rolled back, leaving the database
+            // in the state it was in prior to the load attempt.
+            using (var trxScope = new TransactionScope())
+            {
+                using (var connection = new SqlConnection(ConfigurationManager.ConnectionStrings["itembank"].ConnectionString))
+                {
+                    connection.Open();
+
+                    foreach (var testPackageXml in testPackageXmls)
+                    {
+                        LoadScoringPackage(testPackageXml, connection);
+                    }
+
+                    // Call the UpdateTDSConfigs stored procedure to move the data from the loader/staging tables into the production/"real" tables.
+                    using (var updateConfigsCmd = new SqlCommand("dbo.UpdateTDSConfigs", connection))
+                    {
+                        updateConfigsCmd.CommandType = CommandType.StoredProcedure;
+                        updateConfigsCmd.Parameters.AddWithValue("@doall", 1);
+
+                        updateConfigsCmd.ExecuteNonQuery();
+                    }
+
+                    // Insert records into the OSS_TIS.dbo.CombinationTestMap table
+                    ExecuteInsertSqls(combinationTestMapSqls, connection);
+
+                    // Insert records into the OSS_TIS.dbo.CombinationTestFormMap table
+                    ExecuteInsertSqls(combinationTestFormMapSqls, connection);
+                }
+
+                trxScope.Complete();
+            }
+        }
+
+        [TestMethod]
+        public void ShouldDeleteAnAssessment()
+        {
+            var assessmentController = new AssessmentsController()
+            {
+                Request = new HttpRequestMessage(HttpMethod.Delete, "http://localhost:44444/api/assessments/" + TEST_PACKAGE_KEY)
+            };
+
+            using (var response = assessmentController.Delete(TEST_PACKAGE_KEY))
+            {
+                Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+
+                using (var connection = new SqlConnection(ConfigurationManager.ConnectionStrings["itembank"].ConnectionString))
+                {
+                    connection.Open();
+                    for (var i = 0; i < validationSqls.Length; i++)
+                    {
+                        using (var validationSqlCommand = new SqlCommand(validationSqls[i], connection))
+                        {
+                            validationSqlCommand.CommandType = CommandType.Text;
+                            int result = (int)validationSqlCommand.ExecuteScalar();
+
+                            Assert.AreEqual(0, result);

```

Comment:
This is good, but we need a test that confirms the delete method doesn't accidentally delete other tests. For example, make a test that loads two, and only deletes one test, then makes sure one is still there. (then this test can clean it all up)

Also, maybe I'm missing it, but do you check there is a matching tests to start with before attempting the delete anf confirming it's 0? Maybe the load failed or loaded somewhere else, but this test will still pass.

Reply:
I've updated the integration test(s):
* Now two assessments are loaded into the database before each test executes
* The `ShouldDeleteTheSpecifiedAssessment()` test will delete one of the assessments (the grade 3 MATH assessment) and verify the other one (the grade 11 ELA assessment) still exists
* Added another integration test that will delete the remaining assessment

While I was working on this, I realized the integration tests depended on running in a particular order, so I fixed that too.